packages/pharos/src/components/table/pharos-table.test.ts

```diff
@@ -0,0 +1,145 @@
+import { fixture, expect } from '@open-wc/testing';
+import type { TemplateResult } from 'lit';
+
+import { html } from 'lit/static-html.js';
+
+import type { PharosTable } from './pharos-table';
+import type { PharosLink } from '../link/pharos-link';
+import type { PharosPagination } from '../pagination/pharos-pagination';
+import type { PharosSelect } from '../select/pharos-select';
+
+describe('pharos-table', () => {
+  let component: PharosTable, componentWithPagination: PharosTable;
+  const columns = [
+    {
+      name: 'Item',
+      field: 'item',
+    },
+    {
+      name: 'Filename',
+      field: 'filename',
+    },
+  ];
+  const rowData = [
+    {
+      item: 1,
+      filename: '12345.jpg',
+      expired_date: '2020-1-1',
+      created_on: '2010-1-1',
+      university: 'University of Michigan',
+    },
+    {
+      item: 2,
+      filename: '123456.jpg',
+      expired_date: '2020-1-1',
+      created_on: '2010-1-1',
+      university: 'University of Michigan',
+    },
+  ];
+
+  beforeEach(async () => {
+    component = await fixture(html`
+      <test-pharos-table .columns="${columns}" .rowData="${rowData}" .totalResults="${2}">
+      </test-pharos-table>
+    `);
+
+    componentWithPagination = await fixture(html`
+      <test-pharos-table
+        .columns="${columns}"
+        .rowData="${rowData}"
+        .hidePagination="${false}"
+        .totalResults="${2}"
+        .pageSizeOptions="${[1, 2]}"
+      >
+      </test-pharos-table>
+    `);
+  });
+
+  it('is accessible', async () => {
+    await expect(component).to.be.accessible();
+  });
+
+  it('is accessible with pagination', async () => {
+    await expect(componentWithPagination).to.be.accessible();
+  });
+
+  it('has 3 rows', async () => {
+    const rows = Array.prototype.slice.call(
+      component.renderRoot.querySelectorAll(`tr`)
+    ) as TemplateResult[];
+    expect(rows.length).to.be.eq(3);
+  });
+
+  it('render rows according to page size', async () => {
+    const rows = Array.prototype.slice.call(
+      componentWithPagination.renderRoot.querySelectorAll(`tr`)
+    ) as TemplateResult[];
+    expect(rows.length).to.be.eq(2);
+  });
+
+  it('shows correct page start number according to page size', async () => {
+    let pageNumber = componentWithPagination.renderRoot.querySelector(`.page-number-display`);
+    expect(pageNumber?.innerHTML).contains('1-');
+    expect(pageNumber?.innerHTML).contains('1 of');
+
+    const selectDropdown = componentWithPagination.renderRoot.querySelector(
+      `pharos-select`
+    ) as PharosSelect;
+    selectDropdown['_select'].value = '2';
+    selectDropdown['_select'].dispatchEvent(new Event('change'));
+
+    await componentWithPagination.updateComplete;
+
+    pageNumber = componentWithPagination.renderRoot.querySelector(`.page-number-display`);
+    expect(pageNumber?.innerHTML).contains('1-');
+    expect(pageNumber?.innerHTML).contains('2 of');
+  });
```

Comment:
I'm having trouble making sense of this test. Is there a reason we can't assert the entire set of data that's supposed to be displayed in the pagination?

Reply:
What do you mean by "entire set of data"? I was trying to verify that the `_pageStartNumber` and `_pageEndNumber` are calculated correctly. But they are private to the component so I have to look at the rendered html to verify, i.e the "(Displaying 1-2 of 2)" part.

Reply:
That's kind of what I mean—why can't we test "Displaying 1-2 of 2" directly instead of these multiple, partial assertions?

Reply:
Ah here is the reason! This is what I see after I test the whole sentence
<img width="1017" alt="image" src="https://github.com/ithaka/pharos/assets/38861633/ff2f143c-ef03-49b9-b9ce-349b46963223">

Reply:
Ah, perhaps you can try `innerText` instead of `innerHTML`?

Reply:
Yes it works!!!